### DISCUSSION TITLE
Optimal flight space for the LPS.

### DISCUSSION THREAD

**Discussion by santiagorg2401** [bitcraze/discussions#54] [Q&A]
Hi,
We are re-designing our flight space for the Crazyflies we have, and we are looking for recommendations for top performance, we already have 10 Crazyflies and the LPS system with 8 anchors in TDoA2 mode and their respective tags decks for the UAVs, however, we've experienced several issues with its performance (as in this old post in the forum https://forum.bitcraze.io/viewtopic.php?t=4654&start=20), so, what recommendations would you have for an optimal space for 10 Crazyflies using the LPS system?, as for room requirements, space, electromagnetic interference, even, air conditioner?

Thank you in advance.

**Comment by krichardsson** [comment]
Hi!

Here are some tips and ticks that might help. Unfortunately there is also a certain degree of "magic" involved that we have not managed to figure out. In some setups it seems to be more problems while other similar systems works fine This also goes for time - some days a system might be more noisy without any apparent reason. My assumption is that it is related to the (radio) environment.

1. Place the anchors in the corners of your flight space as described in https://www.bitcraze.io/documentation/tutorials/getting-started-with-loco-positioning-system/. In TDoA2 the order of transmissions from the anchors is fixed and the difference of distance will always be measured in the pairs 0-1, 1-2, 2-3 and so on, hence the recommended order. In TDoA3 the order is randomized and you will probably get measurements between all possible pairs. I'm not sure how big difference it makes though.
2. Place the anchors with the antenna pointing upwards, I believe the radiation pattern is more uniform around this axis (https://www.decawave.com/dwm1000/datasheet/)
3. If your space is "flat", that is the length and/or width is much larger than the height, the estimator might not get enough information in the Z-direction for proper height estimation. In this case you might want to help the estimator by adding a Z-ranger or Flow deck (which also helps in X and Y).
4. Increase the TX power of the anchors. Low level config instructions are here https://www.bitcraze.io/documentation/repository/lps-node-firmware/master/development/anchor-low-level-config/. To increase the TX power, set the desired power and `Force TX power=true`
5. Avoid metal surfaces as they act as mirrors
6. Avoid big bags of water (AKA humans) in the flight space
7. Make sure the battery on the Crazyflie is NOT placed under the UWB antenna, move it a bit backwards a bit to leave the antenna free. 
8. Make sure the anchors have a good power supply, they are a bit picky about voltage and draws quite some current from time to time.
9. The same goes for the Crazyflie, a fresh battery is better than a flat one. When the battery is close to running out things starts to get shaky, both because of powering the UWB module (not a big problem in TDoA as it is only receiving) but also to power the motors that must handle the slightly heavy Crazyflie (due to the extra weight of the Loco deck).

**Reply by santiagorg2401** [reply]
Hi,
Interesting thank you, would you say that it is better a room in a higher floor than a lower one to place the system?, Do you think air conditioning may be an issue for turbulence?, And, what size would you recommend for best performance?

Thanks!

**Comment by krichardsson** [comment]
> would you say that it is better a room in a higher floor than a lower one to place the system?

I'm not exactly sure what you mean but my interpretation is that you are asking if it makes a difference if the system is set up in a room that is on the lower floors of a building (closer to the ground) compared to a room at the top of the building. Assuming the rooms have similar properties, then I can not see that it should make any difference. 

> Do you think air conditioning may be an issue for turbulence?

Yes, air flow can cause some trouble if it is too strong as the Crazyflie has to compensate for it. Similar problems arise if one Crazyflie flies too close above another one when the lower will have to compensate for the down wash.

> what size would you recommend for best performance?

This is a hard question to answer. The default power settings are good up to 2-3 meters and with increased power it works up to 8-10 meters. For even larger spaces it is possible to switch to a long range mode (available in TDoA3) that probably works up to 50-100 meters (I'm not sure as it is slightly experimental). 
As mentioned earlier, it is not possible to get a good Z-estimate in "flat" systems (due to geometrical reasons), so the more "cubic" the set up is, the more similar will the estimation quality be along all axises. Larger systems tend to be flatter due to limitations in ceiling height, but that depends on your room of course. 
Another factor is that the error is about the same, regardless of system size which means that it becomes more apparent or dominant in a smaller system.
A third factor is possibly that it is convenient to be able to reach the anchors without too much complications for maintenance or configuration. If the top anchors are very high up you might need long ladders or similar.
I suppose the most important factor is what your needs are.

**Reply by santiagorg2401** [reply]
Thank you, your insight is appreciated.
